DecipherNow/gm-fabric-dashboard
# 192 vendor bundle

Resolves #192 

## Proposed Changes

  - Generate separate vendor bundles from source code bundles.
  
## How to Test

  - `npm run build`
  - `docker build -t deciphernow/gm-fabric-dashboard -f docker-prod/Dockerfile .`
  - `docker run -it -p 1337:1337 deciphernow/gm-fabric-dashboard`
  - ensure static assets are loaded and app functions properly

## Comparison of bundles
### Before
```
  207.02 KB  build/static/js/main.9cb73c69.js
  77.62 KB   build/static/js/1.0fc61dda.chunk.js
  70.51 KB   build/static/js/5.6810cf62.chunk.js
  70.51 KB   build/static/js/6.40e23795.chunk.js
  69.56 KB   build/static/js/7.29d6c95d.chunk.js
  59.19 KB   build/static/js/3.c4417e06.chunk.js
  41.78 KB   build/static/js/10.36a85888.chunk.js
  41.66 KB   build/static/js/9.679a9e46.chunk.js
  41.52 KB   build/static/js/4.5bc8d9ea.chunk.js
  31.64 KB   build/static/js/8.02687a3c.chunk.js
  21.05 KB   build/static/js/2.3b0e112c.chunk.js
  15.22 KB   build/static/js/11.4ead6138.chunk.js
  1.06 KB    build/static/css/main.927a1cba.css
  1.03 KB    build/static/js/0.3fe7a953.chunk.js
  567 B      build/static/js/12.1790467a.chunk.js
  564 B      build/static/js/13.b8e18f7e.chunk.js
  511 B      build/static/js/14.8783154f.chunk.js
```

### After
```
  163.64 KB              build/static/js/vendor.e226690f.js # new main vendor bundle
  92.35 KB (+91.32 KB)   build/static/js/0.dd0d6fba.chunk.js # new async chunks vendor bundle
  56.97 KB (-150.04 KB)  build/static/js/main.ff1e5bd6.js
  20.07 KB (-1001 B)     build/static/js/2.fbce8b4c.chunk.js
  14.47 KB (-17.17 KB)   build/static/js/8.ba4f0159.chunk.js
  14.47 KB (-27.2 KB)    build/static/js/9.e7d6594b.chunk.js
  13.62 KB (-28.16 KB)   build/static/js/10.a2d00513.chunk.js
  11.8 KB (-58.71 KB)    build/static/js/5.a26c996f.chunk.js
  11.65 KB (-3.56 KB)    build/static/js/11.29ece0d2.chunk.js
  5.63 KB (+5.08 KB)     build/static/js/13.1ea120f8.chunk.js
  5.54 KB (+4.98 KB)     build/static/js/12.73fdbd38.chunk.js
  5.41 KB (+4.92 KB)     build/static/js/14.b08af78a.chunk.js
  5.03 KB (-65.48 KB)    build/static/js/6.84f1f8a4.chunk.js
  4.77 KB (-64.79 KB)    build/static/js/7.2b143cec.chunk.js
  1.06 KB (-76.56 KB)    build/static/js/1.d2a7f126.chunk.js
  1.06 KB                build/static/css/main.927a1cba.css
  950 B                  build/static/js/manifest.c8d39664.js
  610 B                  build/static/js/15.d2d7e62b.chunk.js
  608 B                  build/static/js/16.af9d0872.chunk.js
  549 B                  build/static/js/17.c38adafb.chunk.js
```

<img width="1916" alt="screen shot 2017-12-06 at 12 19 31 am" src="https://user-images.githubusercontent.com/2581080/33645911-30ae6a3c-da1b-11e7-913a-fb97f23a4b2c.png">
